### PR TITLE
Add LED Morse code lesson with basic and complete code and challenges

### PR DESCRIPTION
**Pull request by Oddant1**

### CODE REVIEW

File: book/lessons/morse-led-1/lesson.md [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+
+Standard Morse code has a gap that is 3 dot durations long between characters
+in the same word. Can you add this feature to the code? For example using the
+string "sos" (using '.' as a dot duration rest) we should see:
+
+s...o...s
+
+**Add Support for Spaces**
+
+You may have noticed that we are missing something very importany. We don't
```
Comment (gregcaporaso) [inline book/lessons/morse-led-1/lesson.md:190]: typo: _importany_ -> _important_

File: book/lessons/morse-led-1/lesson.md [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+    'F' : [DOT, DOT, DASH, DOT],
+    'G' : [DASH, DASH, DOT],
+    'H' : [DOT, DOT, DOT, DOT],
+    'I' : [DOT, DOT],
+    'J' : [DOT, DASH, DASH, DASH],
+    'K' : [DASH, DOT, DASH],
+    'L' : [DOT, DASH, DOT, DOT],
+    'M' : [DASH, DASH],
+    'N' : [DASH, DOT],
+    # Fun fact, you may be familiar with "SOS" as a call for help. Some people
```
Comment (gregcaporaso) [inline book/lessons/morse-led-1/lesson.md:247]: Again, `,` -> `:`.

File: book/lessons/morse-led-1/lesson.md [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+    '1' : [DOT, DASH, DASH, DASH, DASH],
+    '2' : [DOT, DOT, DASH, DASH, DASH],
+    '3' : [DOT, DOT, DOT, DASH, DASH],
+    '4' : [DOT, DOT, DOT, DOT, DASH],
+    '5' : [DOT, DOT, DOT, DOT, DOT],
+    '6' : [DASH, DOT, DOT, DOT, DOT],
+    '7' : [DASH, DASH, DOT, DOT, DOT],
+    '8' : [DASH, DASH, DASH, DOT, DOT],
+    '9' : [DASH, DASH, DASH, DASH, DOT],
+    # Fun fact, Morse code didn't originally have a sequence for the period,
```
Comment (gregcaporaso) [inline book/lessons/morse-led-1/lesson.md:276]: Again, `,` -> `:`.

**Review by gregcaporaso — COMMENTED** [review]
This is great @Oddant1, thanks for putting this together! I just have a few minor comments.

File: book/lessons/morse-led-1/lesson.md [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+        # Iterate over the blink durations for the character (found in the
+        # ENCODING dictionary)
+        for duration in ENCODING[character]:
+            # Blink for the duration specified in the dictionary
+            blink(led, duration)
+
+
+# If this file is the one that Python started execution with, run this code.
+# Otherwise it would start executing from the top of the file
+if __name__ == '__main__':Anthony
```
[the remark this answers is not in the document]
Comment (Oddant1) [reply book/lessons/morse-led-1/lesson.md:137]: Yep that is probably what happened. I still sometimes manage to do that

**Comment by Oddant1** [comment]
Sorry about some of those typos @gregcaporaso, it seems like VSCode stopped spellchecking for some reason.

**Review by gregcaporaso — APPROVED** [review]
Thanks for the edits. Again, this looks _great_, nice work!

**Pull request merged**